hw4/submission_template/test/test_ip_addr_file.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+from pathlib import Path
+import os, sys
+parentdir = Path(__file__).parents[1]
+sys.path.append(parentdir)
+
+
+class IPAddrFileTest(unittest.TestCase):
+    def setUp(self):
+        self.ip_file_path = os.path.join(parentdir, 'ip_address.txt')
+
+    def test_tsv_dataset_file(self):
+        print("\nRUNNING TESTS FOR HW4 - ip_address.txt")
+        print("Ensure txt file exists")
+        self.assertEqual(os.path.isfile(self.ip_file_path), True)
+        print("OK")
+
+        
+        with open(self.ip_file_path, 'r', encoding='utf-8') as f:
+            print("Check structure - file should contain only one line...")
+            lines = f.readlines()
+            self.assertEqual(len(lines), 1)
+            print("OK")
+
+            print("Check structure - string should be an IP address")
+            ip_addr = lines[0]
+            print(ip_addr)
+            self.assertRegex(ip_addr, r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}')
+            print("OK")
```

Comment:
We should allow bare hostnames as well.
Perhaps `(((\d{1,3}\.){3}(\d{1,3}))|([a-zA-Z0-9]*\.[a-zA-Z]+))`?

Reply:
We could - my 2 cents: some folks are already having issues following the guidelines. The more options we give, the higher will be the margin for misunderstanding. WDYT @toshokan ?

Reply:
I strongly believe that they should not lose marks for giving a resolvable name instead of an IP if they know how their way around the DNS, but I don't think we have to advertise it as an option (that way we can avoid confusing people). (the same way we don't _technically_ care whether their hosting provider is AWS, or any other that meets the needs of the assignment, but we don't advertise that) 

When grading, the scripts work identically whether it's an IP or resolves to an IP (minus the check for the format), and I think our grading scripts should match the validation tests where possible, so I see this change as just making it easier to test these students rather than having to manually grade them again.